xbmc/platform/win32/filesystem/Win32Directory.cpp

```diff
@@ -81,35 +102,34 @@ bool CWin32Directory::GetDirectory(const CURL& url, CFileItemList &items)
       continue;
     }
 
-    CFileItemPtr pItem(new CFileItem(itemName));
-
-    pItem->SetFolder((findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0);
-    if (pItem->IsFolder())
-      pItem->SetPath(pathWithSlash + itemName + '\\');
-    else
-      pItem->SetPath(pathWithSlash + itemName);
+    const bool isDir = (findData.dwFileAttributes & FILE_ATTRIBUTE_DIRECTORY) != 0;
+    const bool isHidden =
+        (findData.dwFileAttributes & (FILE_ATTRIBUTE_HIDDEN | FILE_ATTRIBUTE_SYSTEM)) != 0 ||
+        itemName.front() == '.'; // mark files starting from dot as hidden
```

Comment:
```suggestion
        itemName.starts_with('.'); // mark files starting from dot as hidden
```

Reply:
How is that different? front() is more consistent with the back/push_back earlier and the push_back you suggest just below

Reply:
It's safe here because of the itemName.empty() test a bit earlier. But good catch!